src/extension/wps/wps-core/src/main/java/org/geoserver/wps/ppio/ProcessParameterIO.java

```diff
@@ -184,8 +184,12 @@ public int compare(ProcessParameterIO o1, ProcessParameterIO o2) {
         }
 
         // load by factory
-        List<PPIOFactory> ppioFactories = GeoServerExtensions
-                .extensions(PPIOFactory.class, context);
+        List<PPIOFactory> ppioFactories;
+        if (context != null) {
+            ppioFactories = GeoServerExtensions.extensions(PPIOFactory.class, context);
```

Comment:
I try to make pull requests small and isolated and changing the behavior or a core method that is probably called hundreds of times throughout the code in order to suppress warning messages in an extension that isn't breaking anything didn't seem like a good idea.  Also, GeoServerExtensions.extensions is called twice within this ProcessParameterIO method (about 5 lines above this change) and all this pull request does is make both calls handle null contexts in the same way.

I agree that GeoServerExtensions should try to use its own context if null is passed to it.  Even if you guys did decide to change GeoServerExtensions.extensions, it would have to be done in a way that doesn't break this pull request since I suspect that this isn't the only place in the code with this extra logic for handling null contexts.